docs/samples/python/customizing_presidio_analyzer.ipynb

```diff
@@ -0,0 +1,677 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the recognizer with the defined pattern\n",
+    "zipcode_recognizer = PatternRecognizer(supported_entity=\"US_ZIP_CODE\", \n",
+    "                                       patterns = [zipcode_pattern],\n",
+    "                                       context= [\"zip\",\"zipcode\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "On the recognizer level, the score would remain 0.01. When being called by the `AnalyzerEngine`, the context words are introduced:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "registry = RecognizerRegistry()\n",
+    "registry.add_recognizer(zipcode_recognizer)\n",
+    "analyzer = AnalyzerEngine(registry=registry)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[type: US_ZIP_CODE, start: 15, end: 20, score: 0.4]"
+      ]
+     },
+     "execution_count": 40,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "analyzer.analyze(text=\"My zip code is 90210\",language=\"en\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The confidence score is now 0.4, instead of 0.01. We can change that by updating `zipcode_recognizer.CONTEXT_SIMILARITY_FACTOR` to something other than 0.4."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example 7: Tracing the decision process\n",
+    "\n",
+    "Presidio-analyzer's decision process exposes information on why a specific PII was detected. Such information could contain:\n",
+    "\n",
+    "- Which recognizer detected the entity\n",
+    "- Which regex pattern was used\n",
+    "- Interpretability mechanisms in ML models\n",
+    "- Which context words improved the score\n",
+    "- Confidence scores before and after each step\n",
+    "And more.\n",
+    "\n",
+    "For more information, refer to the [decision process documentation](https://microsoft.github.io/presidio/analyzer/decision_process/).\n",
+    "\n",
+    "Let's use the decision process output to understand how the zip code value was detected:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'original_score': 0.01,\n",
+      " 'pattern': '(\\\\d{5}(?:\\\\-\\\\d{4})?)$',\n",
+      " 'pattern_name': 'zip code (weak)',\n",
+      " 'recognizer': 'PatternRecognizer',\n",
+      " 'score': 0.4,\n",
+      " 'score_context_improvement': 0.39,\n",
+      " 'supportive_context_word': 'zip',\n",
+      " 'textual_explanation': None,\n",
+      " 'validation_result': None}\n"
+     ]
```

Comment:
Not sure the response should look this way.
![Screen Shot 2021-03-11 at 13 51 55](https://user-images.githubusercontent.com/1862328/110783361-031bc900-8271-11eb-80c1-6f4ccb15704d.png)

Reply:
Same issue as above?

Reply:
yes. Also, shouldn't it presented as code? it will look better...